### ISSUE TITLE
Keep the raw job spec or a SHA1 of the job spec

### ISSUE THREAD

**Issue by cbednarski** [hashicorp/nomad#681]
When tracking a job specification in git or other external system of record, it's useful to be able to see what version of the job is currently running in Nomad.

We should track a SHA1 of the job spec or the entire raw job spec so it can be correlated with a version from version control.

**Comment by mitchellh** [comment]
I'm not sure if this helps but @dadgar has been integrating github.com/mitchellh/hashstructure for computed node classes to unique nodes but we could also easily do that to hash the job definitions as well and compare those. This is what Otto does since it doesn't store the config but can tell you what a config maps to.

**Comment by diptanu** [comment]
Yeah I think hashstructure would be the better than just using SHA1 on the contents of the raw file. We do similar things to know the version of service definitions on a job file vs what we have registered with Consul to know if they need to be updated.

**Comment by schmichael** [comment]
Closing for 2 reasons:

1. Templating your git sha into your jobspec is the suggested workflow that we encoded with the release of [nomad-pack](https://github.com/hashicorp/nomad-pack/blob/c218779d9557cb2677d2ae78584153e878fb4e4a/internal/runner/job/metadata.go#L29)
2. There are a lot of considerations to be weighed for whether or not Nomad should store the raw HCL for a job as this issue implies. If users are interested in that I think it's best discussed in a new focused issue.

(I have to admit it's a little intimidating closing an issue from you three gentlemen, but I think it's safe to assume you all forgot about it 4ish years ago anyway. :sweat_smile: )
